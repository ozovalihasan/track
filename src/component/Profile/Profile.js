import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Profile = ({ handleClick, user }) => (
  <div>
    <div>
      {
        (user.username) ? (
          <div>

            <p>
              Hi
              {' '}
              {user.username}
            </p>
            <button type="button" onClick={handleClick}>
              Logout
            </button>
          </div>
        ) : (
          <div>
            <Link to="/log-in">Login</Link>
            <Link to="/sign-up">Sign up</Link>
          </div>
        )
    }

    </div>
  </div>
);

Profile.propTypes = {
  handleClick: PropTypes.func.isRequired,
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }).isRequired,
};

export default Profile;
