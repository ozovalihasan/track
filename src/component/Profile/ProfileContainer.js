import { useDispatch, useSelector } from 'react-redux';
import {
  resetApp, resetApi, resetPiece, resetTakenTime, resetTrackedItem, userSignOut,
} from '../../redux';
import Profile from './Profile';

const ProfileContainer = () => {
  const dispatch = useDispatch();
  const user = useSelector(state => state.user);

  const handleClick = () => {
    localStorage.removeItem('token');
    dispatch(resetApi());
    dispatch(resetApp());
    dispatch(resetTrackedItem());
    dispatch(resetPiece());
    dispatch(resetTakenTime());
    dispatch(userSignOut());
  };

  return <Profile handleClick={handleClick} user={user} />;
};

export default ProfileContainer;
