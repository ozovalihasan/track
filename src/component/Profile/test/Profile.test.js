import { render, screen } from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter, Redirect, Route, Switch,
} from 'react-router-dom';
import '@testing-library/jest-dom';

import userEvent from '@testing-library/user-event';
import Profile from '../Profile';

const user = { username: 'Mock Username' };

let handleClick;
let renderReadyComponent;

beforeEach(() => {
  handleClick = jest.fn();
  renderReadyComponent = (
    <BrowserRouter>
      <Profile user={user} handleClick={handleClick} />
      <Redirect to="/" />
      <Switch>
        <Route exact path="/" render={() => <>Main Page</>} />
        <Route exact path="/log-in" render={() => <>Log In Page</>} />
        <Route exact path="/sign-up" render={() => <div>Sign Up Page</div>} />
      </Switch>
    </BrowserRouter>
  );
});

afterEach(() => {
  handleClick.mockClear();
});

describe('<Profile />', () => {
  describe('If user.username is not empty', () => {
    it('renders user.username', () => {
      render(renderReadyComponent);

      expect(screen.getByText(/Mock Username/i)).toBeInTheDocument();
    });

    it('calls handleClick when \'Logout\' is clicked', () => {
      render(renderReadyComponent);
      userEvent.click(screen.getByText(/Logout/i));

      expect(handleClick).toHaveBeenCalled();
    });

    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent);
      expect(renderedContainer).toMatchSnapshot();
    });
  });

  describe('If user.username is empty', () => {
    it('redirects \'/log-in\' when \'Login\' is clicked', () => {
      user.username = '';
      render(renderReadyComponent);
      userEvent.click(screen.getByText(/Log in/i));

      expect(screen.getByText(/Log In Page/i)).toBeInTheDocument();
    });

    it('redirects \'/sign-up\' when \'Sign up\' is clicked', () => {
      user.username = '';
      render(renderReadyComponent);
      userEvent.click(screen.getByText(/Sign up/i));

      expect(screen.getByText(/Sign Up Page/i)).toBeInTheDocument();
    });

    it('renders correctly', () => {
      user.username = '';
      const renderedContainer = render(renderReadyComponent);
      expect(renderedContainer).toMatchSnapshot();
    });
  });
});
