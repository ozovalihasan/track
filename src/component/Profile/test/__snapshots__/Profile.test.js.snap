// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> If user.username is empty renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div>
            <a
              href="/log-in"
            >
              Login
            </a>
            <a
              href="/sign-up"
            >
              Sign up
            </a>
          </div>
        </div>
      </div>
      Main Page
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div>
          <a
            href="/log-in"
          >
            Login
          </a>
          <a
            href="/sign-up"
          >
            Sign up
          </a>
        </div>
      </div>
    </div>
    Main Page
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Profile /> If user.username is not empty renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div>
            <p>
              Hi
               
              Mock Username
            </p>
            <button
              type="button"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
      Main Page
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div>
          <p>
            Hi
             
            Mock Username
          </p>
          <button
            type="button"
          >
            Logout
          </button>
        </div>
      </div>
    </div>
    Main Page
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
